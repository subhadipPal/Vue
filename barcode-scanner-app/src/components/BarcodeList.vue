<template>
  <div class="barcode-list-container">
    <div
      class="barcode-list"
      :style="
        barcodeStore.scannedBarCodes.length === 0
          ? 'display: none'
          : 'display: block'
      "
    >
      <v-list lines="one">
        <template
          v-for="(item, index) in barcodeStore.scannedBarCodes"
          :key="item"
        >
          <div class="barcode-list-item">
            <v-list-item
              :title="item"
              style="width: 80%; text-align: start"
            ></v-list-item>
            <v-btn variant="plain" @click="removeScannedCode(index)"
              >Delete</v-btn
            >
          </div>
          <v-divider
            v-if="index !== barcodeStore.scannedBarCodes.length - 1"
            :key="index"
          ></v-divider>
        </template>
      </v-list>
    </div>
    <span
      :style="
        barcodeStore.scannedBarCodes.length === 0
          ? 'display: block'
          : 'display: none'
      "
      >No barcodes yet</span
    >
  </div>
</template>
<script lang="ts">
import { defineComponent } from "vue";
import useScannedBarCodesStore from "@/stores/scannerStore";

export default defineComponent({
  name: "BarcodeList",
  methods: {
    removeScannedCode(index: number) {
      this.barcodeStore.spliceBarCodes(index);
    },
  },
  setup() {
    const barcodeStore = useScannedBarCodesStore();
    return {
      barcodeStore,
    };
  },
});
</script>
<style lang="scss">
.barcode-list-container {
  margin-top: 50px;
}
.barcode-list {
  margin: auto 200px;
  border: 1px solid lightgrey;
  border-radius: 10px;
  padding: 10px;
}
.barcode-list-item {
  display: flex;
  flex-direction: row;
}
</style>
