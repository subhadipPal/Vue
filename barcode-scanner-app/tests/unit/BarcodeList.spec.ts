import { mount, enableAutoUnmount, VueWrapper } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";
import useScannedBarCodesStore from "@/stores/scannerStore";

import BarcodeList from "../../src/components/BarcodeList.vue";
describe("HelloWorld.vue", () => {
  let wrapper: VueWrapper<any>, vuetify;
  beforeEach(() => {
    vuetify = {
      mocks: {
        $vuetify: {
          lang: {
            t: (val: string) => val,
          },
        },
      },
    };

    wrapper = mount(BarcodeList, {
      global: {
        plugins: [createTestingPinia()],
      },
      vuetify,
    });
  });
  enableAutoUnmount(afterEach);

  it("renders BarCodeList for scanned codes", () => {
    const barCodeStore = useScannedBarCodesStore();
    barCodeStore.$patch({
      scannedCodes: ["11111111", "222222", "333333"],
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
