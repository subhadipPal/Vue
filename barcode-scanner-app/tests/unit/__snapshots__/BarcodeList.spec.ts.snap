// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders BarCodeList for scanned codes 1`] = `
<div class="barcode-list-container">
  <div class="barcode-list" style="display: none;">
    <v-list lines="one"></v-list>
  </div><span style="display: block;">No barcodes yet</span>
</div>
`;
